hpc/jvergeyn-vacc-strategies.py

```diff
@@ -0,0 +1,681 @@
+
+df_sciensano = sciensano.get_sciensano_COVID19_data(update=False)
+df_sciensano['D_cum'] = df_sciensano.D_tot.cumsum()
+df_google = mobility.get_google_mobility_data(update=False, plot=False)
+
+
+# ## Load posterior parameter distributions of calibrated parameters
+
+# Load samples dictionaries
+with open('../data/interim/model_parameters/COVID19_SEIRD/calibrations/national/BE_4_prev_full_2021-01-30_WAVE2_GOOGLE.json', 'r') as fp:
+    samples_dict = json.load(fp)
+
+with open('../data/interim/model_parameters/COVID19_SEIRD/calibrations/national/initial_states_2020-09-01.json', 'r') as fp:
+    initial_states_sept = json.load(fp) 
+    
+# Set initial states of VE (vaccine elegible)
+initial_states_sept['VE'] = (np.array(initial_states_sept['S'])+
+                             np.array(initial_states_sept['R'])+
+                             np.array(initial_states_sept['E'])+
+                             np.array(initial_states_sept['I'])+
+                             np.array(initial_states_sept['A'])).squeeze()
+
+## Scenarios
+
+start_sim = '2020-09-01'
+end_sim = '2021-09-01'
+start_fig = '2021-01-01'
+warmup = 0
+n_samples = 100
+n_draws_per_sample = 1000
+
+Re_1feb = 0.958*1.4
+# current estimation of UK prevalence between 20 and 40% => take 30%
+current_UK = 0.3
+# 5-6 days incubation
+incubation_period = 5
+n_periods = 30/incubation_period
+# How much on January 1? (30 days ago)
+portion_new_strain_introduced = current_UK/(Re_1feb**n_periods) #0.001
+injection_day = (pd.Timestamp('2021-01-01') - pd.Timestamp(start_sim))/pd.Timedelta('1D')
+
+UL = 0.975
+LL = 0.025
+
+# doses per day (Regeringscommissariaat Corona, published by De Tijd)
+d = {}
+d['jan'] = 31765
+d['feb'] = 45897
+d['mar-apr'] = 128499
+d['may-aug'] = 78358
+
+# proportion of age group living in nursing homes
+# for simplicity we will use the data for respectively age groups 60+, 70+ and 80+
+NH = {}
+NH['65+'] = 0.007
+NH['75+'] = (0.05+0.029)/2
+NH['85+'] = (0.137+0.267)/2
+
+## Helpfunction vaccination
+
+def smoothclamp(x, mi, mx): 
+    return mi + (mx-mi)*(lambda t: np.where(t < 0 , 0, np.where( t <= 1 , 3*t**2-2*t**3, 1 ) ) )( (x-mi)/(mx-mi) )
+
+def smooth_dose(dose, sense, start_dose=0, end_dose=0, N=1000):
+    if sense == 'up':
+        x = np.linspace(start_dose, dose, num=N)
+        return smoothclamp(x, start_dose, dose)
+    elif sense == 'down':
+        x = np.linspace(dose, end_dose, num=N)
+        return smoothclamp(x, dose, end_dose)
+    else:
+        raise ValueError(
+                "sense must be 'up' or 'down' "
+            )
+
+def smooth_dose_ts(daily_dose, start_t, end_t, spread, start_dose=0, end_dose=0, N=1000):
+    """
+    max_to_vacc : max number to vaccinate
+    daily_dose : daily number of vaccinations
+    spread : number of days at start and end to spread out step function
+    N : resolution of the smoothed step function
+    start_dose : level to start smoothing from (0 or previous daily dose)
+    
+    """
```

Comment:
Kan je verder ook nog een uitleg toevoegen aan de functies? Ik versta niet 100% wat ze doen. Ze smoothen de hoeveelheden doses die je doorgeeft om stijfheid in de solver te vermijden? Kan je die uitleg ook toevoegen, dat maakt het makkelijker om in te pikken.